src/AppKit/Compat.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.ComponentModel;
+
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.AppKit{
+#if !XAMCORE_4_0
+	public partial class NSCollectionViewTransitionLayout {
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("init")]
+		public NSCollectionViewTransitionLayout () : base (NSObjectFlag.Empty)
```

Comment:
I tried to follow your suggestion here: https://github.com/xamarin/xamarin-macios/pull/2245#discussion_r124869387

You are correct that I forgot the obsolete. Adding.